src/runtime_src/core/common/sensor.cpp

```diff
@@ -250,22 +249,31 @@ read_data_driven_thermals(const std::vector<xq::sdm_sensor_info::data_type>& out
 }
 
 static ptree_type
-read_data_driven_mechanical(std::vector<xq::sdm_sensor_info::data_type> output)
+read_data_driven_mechanical(std::vector<xq::sdm_sensor_info::data_type>& output,
+                            std::vector<xq::sdm_sensor_info::data_type>& temp)
 {
   ptree_type root;
   ptree_type pt;
   ptree_type fan_array;
 
   // iterate over output data, store it into property_tree
-  for(const auto& tmp : output)
+  for (const auto& tmp : output)
   {
     pt.put("location_id", tmp.label);
     pt.put("description", tmp.label);
     pt.put("speed_rpm", tmp.input);
-    pt.put("critical_trigger_temp_C", "N/A");
     pt.put("is_present", "true");
-    fan_array.push_back({"", pt});
   }
+  // iterate over output data, store it into property_tree
+  for (const auto& tmp : temp)
```

Comment:
Earlier, Soren had recommend a code change, which you did.  
Question is: Why didn't you apply it to ALL the code you changed?